### PR TITLE
feat(suite): ignore FW hash check unrecognized-version

### PR DESCRIPTION
**Pull request by Lemonexe**
## Description

Currently, FW hash check [returns `'unknown-release'` error](https://github.com/trezor/trezor-suite/blob/aeab689d34d2d36b679d88fd4ca8dbe77b86be79/packages/connect/src/device/Device.ts#L787) if the FW version is not _locally_ known. A countefeit device might be like that, so this was considered a "hard failure".

:arrow_right: ignore this case in FW hash check, because FW revision check handles it more elegantly - it solves edge cases described [here](https://github.com/trezor/trezor-suite-private/issues/115) and [here](https://github.com/trezor/trezor-suite-private/issues/117)

## Related Issue

Part of [private#117](https://github.com/trezor/trezor-suite-private/issues/117)

Spoils PRs #15059, #15107

## QA instructions

- open suite, make sure in Settings > Device > Danger zone: that Firmware revision check is **on**
- use with trezor-user-env `2-main` or `1-main`
- you should get Ghost modal :ghost:  & banner
"Firmware revision check failed. Your Trezor may be counterfeit."
- in Settings > Debug > Device: turn **off** only Check firmware Revision, make sure Check firmware Hash is **on**,
- reload suite, should be clear :heavy_check_mark: 
- use with trezor-user-env latest official release for any device
- you should get Ghost modal :ghost:  & banner
"Firmware hash check failed. Your Trezor may be counterfeit."
- restart suite with any legit physical device with latest fw 
- make sure in Settings > Device > Danger zone: that Firmware revision check is **on**
- should be clear :heavy_check_mark:

### CODE REVIEW

File: packages/suite/src/components/suite/banners/SuiteBanners/FirmwareRevisionCheckBanner.tsx [hunk 1]
```diff
@@ -9,44 +9,33 @@ import { useSelector } from 'src/hooks/suite';
...
-    selectIsUnrecognizedFirmwareWithOutdatedSuite,
 } from 'src/reducers/suite/suiteReducer';
+import { skippedHashCheckErrors } from 'src/constants/suite/firmware';
 
 const revisionCheckMessages: Record<FirmwareRevisionCheckError, TranslationKey> = {
     'cannot-perform-check-offline': 'TR_DEVICE_FIRMWARE_REVISION_CHECK_UNABLE_TO_PERFORM',
     'other-error': 'TR_FIRMWARE_REVISION_CHECK_OTHER_ERROR',
     'revision-mismatch': 'TR_FIRMWARE_REVISION_CHECK_FAILED',
-    'firmware-version-unknown': 'TR_FIRMWARE_REVISION_CHECK_FAILED',
+    'firmware-version-unknown': 'TR_DEVICE_FIRMWARE_HASH_CHECK_UNKNOWN_RELEASE', // TODO just for testing
```
Comment (peter-sanderson) [inline packages/suite/src/components/suite/banners/SuiteBanners/FirmwareRevisionCheckBanner.tsx]: todo shall stay?

Comment (Lemonexe) [reply packages/suite/src/components/suite/banners/SuiteBanners/FirmwareRevisionCheckBanner.tsx]: :see_no_evil: thanks for catching, fixed in 38237742

**Review by peter-sanderson — APPROVED** [review]

File: packages/suite/src/constants/suite/firmware.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { FirmwareHashCheckError, FirmwareRevisionCheckError } from '@trezor/connect';
+
+export const skippedRevisionCheckErrors = [
```
Comment (peter-sanderson) [inline packages/suite/src/constants/suite/firmware.ts:3]:
```suggestion
export const skippedRevisionCheckErrors: FirmwareRevisionCheckError[] = [''

... as const
```
works?

File: packages/suite/src/constants/suite/firmware.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    'cannot-perform-check-offline',
+    'other-error',
+] satisfies FirmwareRevisionCheckError[];
+
+export const skippedHashCheckErrors = [
```
Comment (peter-sanderson) [inline packages/suite/src/constants/suite/firmware.ts:8]:
```suggestion
export const skippedHashCheckErrors: FirmwareHashCheckError[] = [
```

File: packages/suite/src/constants/suite/firmware.ts (hunk 2 above)
Comment (Lemonexe) [reply packages/suite/src/constants/suite/firmware.ts:3]: Discussed in person, resolving.

File: packages/suite/src/constants/suite/firmware.ts (hunk 3 above)
Comment (Lemonexe) [reply packages/suite/src/constants/suite/firmware.ts:8]: Discussed in person, resolving.

**Pull request merged**